test/user_search/active_directory_test.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../test_helper'
+require 'mocha/mini_test'
+
+class GitHubLdapActiveDirectoryUserSearchTests < GitHub::Ldap::Test
+
+  def setup
+    @ldap = GitHub::Ldap.new(options)
+    @ad_user_search = GitHub::Ldap::UserSearch::ActiveDirectory.new(@ldap)
+  end
+
+  def test_global_catalog_returns_empty_array_for_no_results
+    mock_global_catalog_connection = Object.new
+    mock_global_catalog_connection.expects(:search).returns(nil)
+    Net::LDAP.expects(:new).returns(mock_global_catalog_connection)
```

Comment:
Could you stub the `@ad_user_search.global_catalog_connection` method instead of `Net::LDAP.new`?